### PR TITLE
[Container Registry] Acr delete tag

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_repository_client.py [hunk 1]
```diff
@@ -63,15 +62,15 @@ def delete_registry_artifact(self, digest: str, **kwargs) -> None:
         """
         raise NotImplementedError("Has not been implemented")
 
-    def delete_tag(self, tag: str, **kwargs) -> None:
+    async def delete_tag(self, tag: str, **kwargs) -> None:
```
Comment (iscai-msft) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_repository_client.py:65]: nit: "Delete a tag from a repository"

File: sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        prev = None
+        async for repo in client.list_repositories():
+            count += 1
+            assert isinstance(repo, six.string_types)
+            assert prev != repo
+            prev = repo
+
+        assert count > 0
+
+    @pytest.mark.skip("Don't want to delete for now")
```
Comment (iscai-msft) [inline sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py:50]: is this going to stay skipped?

**Review by iscai-msft — APPROVED** [review]

File: sdk/containerregistry/azure-containerregistry/tests/recordings/test_container_registry_client_async.test_delete_repository_does_not_exist.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      content-type: application/json; charset=utf-8
+      date: Thu, 01 Apr 2021 17:31:49 GMT
+      server: openresty
+      strict-transport-security: max-age=31536000; includeSubDomains
+      transfer-encoding: chunked
+      x-ms-ratelimit-remaining-calls-per-second: '333.316667'
+    status:
+      code: 200
+      message: OK
+    url: https://seankane.azurecr.io/oauth2/exchange
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/tests/recordings/test_container_registry_client_async.test_delete_repository_does_not_exist.yaml:58]: maybe open an issue to scrub this?

File: sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        client = self.create_registry_client(containerregistry_baseurl)
+
+        repositories = client.list_repositories()
+
+        count = 0
+        prev = None
+        async for repo in client.list_repositories():
+            count += 1
+            assert isinstance(repo, six.string_types)
+            assert prev != repo
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py:45]: is there a concern for the service returning dup items?

**Review by kristapratico — APPROVED** [review]
🚀

File: sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py (hunk 2 above)
Comment (seankane-msft) [reply sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py:50]: No, I can fix that now actually and I will before merging.

File: sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py (hunk 4 above)
Comment (seankane-msft) [reply sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py:45]: No, wanted to verify the deserialization.

**Pull request merged**